### ISSUE TITLE
[Due for payment 2025-02-10] Revert #52569

### ISSUE THREAD

**Issue by JmillsExpensify** [Expensify/App#54054]
After giving the solution in https://github.com/Expensify/App/issues/52569 a try, we've realized that it creates a confusing situation in the Inbox, which is:

 - Workspace chat is GBRed
 - Report has a grey `Approved` button

So in short, our solution for `approve` and `pay` (specifically with a report that has some held expenses and some unheld expenses) ends up breaking GBR. 

As a result, we're going to revert the PR from #52569. That restores GBR to the `Inbox` and workspace chat. In addition, for the Search page specifically, we are going to disable the `Approve` button for this case, just like we landed on doing for `Pay`.

<details><summary>Issue Owner</summary>Current Issue Owner: @JmillsExpensify</details>

**Comment by mananjadhav** [comment]
@JmillsExpensify The expectation is that we revert the PR, and then disable Approve button as well as Pay as a new requirement? Would it be a single PR or two PRs - one revert and then second to disable Approve?

I can help with the review on this one if needed.

**Comment by luacmartins** [comment]
I think we want to:
1. Revert the behavior of changing the button colors. 
2. On Search, disable the option in the bulk action dropdown if any selected items are on hold

**Comment by mananjadhav** [comment]
Yes, I can pick this.

**Comment by luacmartins** [comment]
Great! All yours ❤️

**Comment by luacmartins** [comment]
@mananjadhav how's this one going? Do you have an ETA for the PR?

**Comment by mananjadhav** [comment]
I am a bit sick, I'll have the Pr by tomorrow.

**Comment by luacmartins** [comment]
Ah no worries. Hope you feel better soon.

**Comment by mananjadhav** [comment]
Apologies I was unwell. I am just getting back and will have this ready in 1-2 days.

**Comment by mananjadhav** [comment]
Will raise the PR today.

**Comment by mananjadhav** [comment]
> I think we want to:
> 
> 1. Revert the behavior of changing the button colors.

I tried doing a straight revert and it didn't work. What I've done is reverted the changes of the props. We also added a prop for success style, do we want to revert that as well @luacmartins?


> 2. On Search, disable the option in the bulk action dropdown if any selected items are on hold

The current behavior is that the option is hidden. Are yo saying we want to add the option and keep it disabled?

https://github.com/user-attachments/assets/9e7108e2-39ad-4a55-9cd4-a5ba1367d4e5

**Comment by luacmartins** [comment]
>I tried doing a straight revert and it didn't work. What I've done is reverted the changes of the props. We also added a prop for success style, do we want to revert that as well @luacmartins?

Yes, let's remove the prop as well. Ee currently only seem to use it if there are held expenses, which is unnecessary now that we want to change that behavior to always show it in success style.

>The current behavior is that the option is hidden. Are yo saying we want to add the option and keep it disabled?

Yes, we should add the option to the dropdown and disable it if the user can't take the action

**Comment by luacmartins** [comment]
@parasharrajat assigned you as C+

**Comment by mananjadhav** [comment]
@luacmartins @parasharrajat Are you aware why do we disable the whole dropdown button if the selectedItem is disabled.

https://github.com/Expensify/App/blob/04cd7b08f2151ff4a42fbe32dc17226c4a972c69/src/components/ButtonWithDropdownMenu/index.tsx#L133

I think we should remove the `!!selectedItem?.disabled` especially because `options.length > 1`.

https://github.com/user-attachments/assets/2b8042f0-3d23-40b7-adcf-7cc59a915623

**Comment by luacmartins** [comment]
I'm not sure. I think it'd make sense to disable the dropdown only if all options are disabled, not only the selected one.

**Comment by mananjadhav** [comment]
Okay thanks for the clarification.

**Comment by mananjadhav** [comment]
@luacmartins I don't think `isAnyTransactionOnHold` would apply to `Delete`, `Hold` and `Unhold` right? 

Or are you suggesting we update the condition respective to the buttons?

For example,
```
const shouldShowHoldOption = !isOffline && selectedTransactionsKeys.every((id) => selectedTransactions[id].canHold);
```
we remove `selectedTransactionsKeys.every((id) => selectedTransactions[id].canHold);` and always show `Hold` and disabled when `selectedTransactionsKeys.every((id) => !selectedTransactions[id].canHold);`

**Comment by luacmartins** [comment]
Ah no, only Approve and Pay are affected by the holds. Ignore my previous comment

**Comment by mananjadhav** [comment]
@parasharrajat The PR in draft is ready for review. I still need to fix the lint issues and add screenshots. But it's in a shape to review. I am working on limited capacity today.

**Comment by luacmartins** [comment]
@parasharrajat it seems like the PR is still in draft 🤔 Am I missing somehting?

**Comment by mananjadhav** [comment]
I am working on the checklist. There are too many lint errors. Trivial ones are fixed, but some would need more time to update methods and then test.

**Comment by luacmartins** [comment]
@mananjadhav I think we can address the remaining ESLint errors in a follow up. They seem unrelated to your PR and some of them can get quite tricky.

**Comment by mananjadhav** [comment]
Okay, how do we then ignore these? Will it affect other PRs? Because otherwise the PR is ready, left with just screenshots.

**Comment by luacmartins** [comment]
I think we can just merge it as is. It'll only affect other PRs touching those files. We should create a follow up PR to address those issues though, but I think it's cleaner that way to not distract from the main point of the current PR.

**Comment by mananjadhav** [comment]
@luacmartins I can see https://github.com/Expensify/App/issues/54657 is marked on hold for this issue. Do you want me to fix it here?

**Comment by luacmartins** [comment]
Yea, let's fix those here. It's a small style issue.

**Comment by mananjadhav** [comment]
Only Approve and Pay right?

**Comment by luacmartins** [comment]
I think we should fix any button that doesn't show a visual cue after being selected and hovered. We have two possible styles for those, they are either gray or green, so we should check both.

**Comment by mananjadhav** [comment]
@luacmartins @Expensify/design Wanted to quickly check what style should I use for `View` and `Review` because when the item is hovered, the color is same as the button color if [check](https://github.com/Expensify/App/issues/54054#issuecomment-2591338757) the video

**Comment by dubielzyk-expensify** [comment]
Hmm. I'm a bit unsure what's going on here cause don't we have this working on prod today? We make it one shade darker than it is which is using the `button-hover` color:

https://github.com/user-attachments/assets/7e942325-d5cf-4df3-a9dd-09318d7ae8b8

Sorry if I've missed something here

**Comment by mananjadhav** [comment]
@dubielzyk-expensify the expectation from the linked issue is when the row is selected and the button is hovered it should be interactive (ie be able to change color). If you check my previous video for Approve button you’ll see the difference.

**Comment by dubielzyk-expensify** [comment]
Can we used the `button-pressed` for hover? 

![Image](https://github.com/user-attachments/assets/517902b0-ab72-4ccb-95ce-da01b4f9cfff)

**Comment by mananjadhav** [comment]
I am afk. But the second row color seems to be different than the current view. The on hover design seems to be the same in the current implementation.

**Comment by shawnborton** [comment]
Yeah, I agree with Jon's comments above - we already change the button BG color when the row is selected, we just need to add a hover style to it - and I like Jon's suggestion of just using the pressed color we have.

**Comment by mananjadhav** [comment]
I am left with this final button color change. Will get to it as soon as I can come online. The rest is done.

**Comment by dannymcclain** [comment]
Agree with Shawn and Jon—that solution makes sense to me.

**Comment by mananjadhav** [comment]
Added the changes.

https://github.com/user-attachments/assets/cbafcbfa-e901-4861-b7fc-750f88b41c70

**Comment by luacmartins** [comment]
Nice! Thanks. It seems like we can get started with the review @parasharrajat

**Comment by mananjadhav** [comment]
@JmillsExpensify quick bump on the message and this is ready for payout.

**Comment by mananjadhav** [comment]
@JmillsExpensify Waiting for the payout summary on this one.

**Comment by luacmartins** [comment]
Just waiting on payment

**Comment by sakluger** [comment]
Summarizing payment on this issue:

Contributor+: @mananjadhav $250, please request on NewDot
Contributor+: @parasharrajat $250, please request on NewDot

**Comment by garrettmknight** [comment]
$250 approved for @mananjadhav

**Comment by mananjadhav** [comment]
Thanks @sakluger @garrettmknight for taking care of this.

**Comment by mananjadhav** [comment]
Once @parasharrajat's payout is approved, we can close this one out.

**Comment by luacmartins** [comment]
We're done!

**Comment by JmillsExpensify** [comment]
$250 approved for @parasharrajat
